### DISCUSSION TITLE
V3 Documents TS definitions

### DISCUSSION THREAD

**Discussion by scerelli** [sanity-io/sanity#3848] [Sanity Studio Support]
Hi, 
i'm trying to migrate to V3 and I was looking for a way, if there is one, to use the documents type definitions via import, for example how can I achieve something like this:

in sanity
```
export const ProductSchema = defineType({
  name: 'product',
  title: 'Product',
  type: 'document',
  ....
```
frontend

`function normalizeProduct(product: ProductSchema <-- this one) {}`

it would be possible to achieve somehting like this with v3 out of the box? like prisma does. because in v2 i was using a type generator but now that api are fully typed i tought you would expose also document definitions types.

**Comment by simonhrogers** [comment]
Ah yes great to know isn’t me being a stupid typescript noob; I was thinking this would be possible and very helpful. It’s surely a selling point of the unified studio + frontend.
